src/components/EmojiPicker/EmojiPickerMenu/BaseEmojiPickerMenu.js

```diff
@@ -0,0 +1,146 @@
+    isFiltered: PropTypes.bool.isRequired,
+
+    /** Array of header emojis */
+    headerEmojis: PropTypes.arrayOf(PropTypes.shape(emojiPropTypes)).isRequired,
+
+    /** Function to scroll to a specific header in the emoji list */
+    scrollToHeader: PropTypes.func.isRequired,
+
+    /** Style to be applied to the list wrapper */
+    listWrapperStyle: stylePropTypes,
+
+    /** Reference to the emoji list */
+    forwardedRef: refPropTypes,
+
+    /** The data for the emoji list */
+    data: PropTypes.arrayOf(PropTypes.shape(emojiPropTypes)).isRequired,
+
+    /** Function to render each item in the list */
+    renderItem: PropTypes.func.isRequired,
+
+    /** Extra data to be passed to the list for re-rendering */
+    // eslint-disable-next-line react/forbid-prop-types
+    extraData: PropTypes.any,
+
+    /** Array of indices for the sticky headers */
+    stickyHeaderIndices: PropTypes.arrayOf(PropTypes.number),
+
+    /** Whether the list should always bounce vertically */
+    alwaysBounceVertical: PropTypes.bool,
+};
+
+const defaultProps = {
+    listWrapperStyle: [],
+    forwardedRef: () => {},
+    extraData: [],
+    stickyHeaderIndices: [],
+    alwaysBounceVertical: false,
+};
+
+/**
+ * Improves FlashList's recycling when there are different types of items
+ * @param {Object} item
+ * @returns {String}
+ */
+const getItemType = (item) => {
+    // item is undefined only when list is empty
+    if (!item) {
+        return;
+    }
+
+    if (item.name) {
+        return CONST.EMOJI_PICKER_ITEM_TYPES.EMOJI;
+    }
+    if (item.header) {
+        return CONST.EMOJI_PICKER_ITEM_TYPES.HEADER;
+    }
+
+    return CONST.EMOJI_PICKER_ITEM_TYPES.SPACER;
+};
+
+/**
+ * Return a unique key for each emoji item
+ *
+ * @param {Object} item
+ * @param {Number} index
+ * @returns {String}
+ */
+const keyExtractor = (item, index) => `emoji_picker_${item.code}_${index}`;
+
+function BaseEmojiPickerMenu({headerEmojis, scrollToHeader, isFiltered, listWrapperStyle, forwardedRef, data, renderItem, stickyHeaderIndices, extraData, alwaysBounceVertical}) {
+    const styles = useThemeStyles();
+    const {translate} = useLocalize();
+
+    return (
+        <>
+            {!isFiltered && (
+                <CategoryShortcutBar
+                    headerEmojis={headerEmojis}
+                    onPress={scrollToHeader}
+                />
+            )}
+            <View style={listWrapperStyle}>
+                <FlashList
+                    ref={forwardedRef}
+                    keyboardShouldPersistTaps="handled"
+                    data={data}
+                    renderItem={renderItem}
+                    keyExtractor={keyExtractor}
+                    numColumns={CONST.EMOJI_NUM_PER_ROW}
+                    stickyHeaderIndices={stickyHeaderIndices}
+                    ListEmptyComponent={<Text style={[styles.textLabel, styles.colorMuted]}>{translate('common.noResultsFound')}</Text>}
```

Comment:
@adhorodyski can you please review [this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1681459454343289) for context and then let us know if you still think we should make a change here?

Reply:
@adhorodyski I applied the following change after internal discussion, could you please share your thoughts on this here as well?

Reply:
Sure thing!
Since Flashlist extends from a PureComponent, by passing in `EmptyComponent={<Text />}` instead of `EmptyComponent={Empty}` we create a new React element with each render, thus breaking the memoisation (a new object reference - in this case `<Text />` behaves exactly like passing in a lambda expression by resulting in a React.createElement call, see https://react.dev/reference/react/createElement).

This is bad performance wise - we not only allocate more memory (N nodes VS 1 node, more GC needed), but also force a rerender on our list - all the comparisons that were made (CPU cycles spent on comparing the props) were a waste. 

The optimisation efforts (like wrapping other props with useCallback etc) lead to nothing if we break the memoisation by passing in a reference to a new object for a single property instead of reusing a component instance.

@roryabraham please let me know if this makes things more clear or I should rephrase something! Happy to chat about it 🙂

Reply:
TLDR: for pure/memoised components `<Text />` behaves like `() =>` called internally - a new object (element) is getting returned.

Reply:
I was confused by this thread, as typically you can't _just like that_ exchange `<Foo>` and `Foo` in React, but then I realized that the type of `ListEmptyComponent` is a union; it accepts both.

```ts
ListEmptyComponent?: React.ComponentType<any> | React.ReactElement<any, string | React.JSXElementConstructor<any>>;
```

Reply:
> a new object (element) is getting returned.

It seems that in this case, we wouldn't trigger extraneous renders, but I guess we would put more stress on the diffing system

Reply:
ok, I think that makes sense. Let's change it back to pass a reference to the component instead of an element directly then.